doc/tag_parser.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import print_function
+import sys
+import os
+from collections import defaultdict
+import fnmatch
+
+# mask for files to parse
+file_masks = {'*.py', '*.focs.txt', '*.macros'}
+
+# mask for files to exclude from parse, checked after file_masks
+exclude_masks = {'__*'}
+
+# prefix to initiate a match condition
+tag_comment = "@content_tag{"
+
+# Root to search for files
+source_dir = sys.argv[1]
+
+# Source and target paths for file generation
+doc_dir = os.path.join(source_dir, 'doc')
+dox_source = os.path.join(doc_dir, 'all_tags.dox.in')
+dox_target = os.path.join(doc_dir, 'all_tags.dox')
+
+# match condition in doc_target to replace with result of parsing
+dox_variable = "${CONTENT_DOCUMENTATION}"
+
+# prefix to source links (being the cpp api, internal links do not exist)
+base_link_url = 'https://github.com/freeorion/freeorion/blob/master/'
+
+
+class SourceInfo(object):
+    """File name and line number"""
+
+    def __init__(self, _name, _line):
+        """
+
+        :type _name: str
+        :type _line: int
+        """
+        assert isinstance(_name, str)
+        self.name = _name
+        assert isinstance(_line, int)
+        self.line = _line
+
+    def __repr__(self):
+        return str.format("{0}#L{1}", self.name, self.line)
+
+
+class TagDescription(object):
+    """Container for a description and source of tag documentation"""
+
+    def __init__(self, _file_name, _line_number, _description):
+        """
+
+        :rtype: object
+        :type _file_name: str
+        :type _line_number: int
+        :type _description: str
+        """
+        assert isinstance(_file_name, str)
+        assert isinstance(_line_number, int)
+        self.file = SourceInfo(_file_name, _line_number)
+        assert isinstance(_description, str)
+        self.description = _description
+
+    def __repr__(self):
```

Comment:
Wrong use of `__repr__`.

https://docs.python.org/2/library/functions.html#repr